### PR TITLE
Python Action Logic

### PR DESCRIPTION
**Pull request by kchason**
- Switch to Python for entrypoint
- Switch to library call with `case-utils==0.12.0`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,21 +4,21 @@ FROM python:3.10-slim-bullseye
...
+ENV CASE_EXTENSION_FILTER ""
 ENV CASE_VALIDATE_ABORT "false"
 
-# Define the command to run the entrypoint.sh script that will detect the type
+# Define the command to run the entrypoint.py script that will detect the type
 # of the path that was provided, apply the filter extension (if appropriate) and
-# run the `case_validate` command against the CASE file(s) to be validated.
-CMD ["bash", "/opt/workspace/entrypoint.sh"]
+# run the `case_validate` function against the CASE file(s) to be validated.
+CMD ["python", "/opt/workspace/entrypoint.py"]
```
Comment (ajnelson-nist) [inline Dockerfile:24]: `python3`?

Comment (kchason) [reply Dockerfile:24]: It's a Python 3.10 image so it only has Python 3 in it

Comment (kchason) [reply Dockerfile:24]: But I can change it if you think it's clearer to review

File: entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import os
+import sys
+
+from case_utils.case_validate import validate
+from case_utils.case_validate.validate_types import ValidationResult
+
+# Get the environment variables as the settings for the validation
+case_version: str = os.environ.get("CASE_VERSION", "case-1.2.0")
```
Comment (ajnelson-nist) [inline entrypoint.py:8]: This seems like a bit of a deep point to bury a hard-coded version.  However, if the intent is to provide the latest ontology version provided by `case-utils`, there is a variable for that:

```python
case_utils.ontology.version_info.CURRENT_CASE_VERSION
```

**Review by ajnelson-nist — APPROVED** [review]
I see no blockers here.

File: entrypoint.py (hunk 2 above)
Comment (kchason) [reply entrypoint.py]: Updated

File: Dockerfile (hunk 1 above)
Comment (ajnelson-nist) [reply Dockerfile]: I'm aware that most virtual environments map `python` to `python3.whichever`, and I'm not surprised that Docker does the same.  My preference is to have the firmer declaration that `python2` isn't even a question.

Comment (kchason) [reply Dockerfile]: Updated

**Pull request merged**